contracts/SpokePoolV3Periphery.sol

```diff
@@ -163,12 +167,17 @@ contract SpokePoolV3Periphery is Lockable, MultiCaller {
      * is the same across networks with different addresses for the wrapped native token and this network's
      * corresponding spoke pool contract. This is to allow this contract to be deterministically deployed with CREATE2.
      */
-    function initialize(V3SpokePoolInterface _spokePool, WETH9Interface _wrappedNativeToken) external nonReentrant {
+    function initialize(
+        V3SpokePoolInterface _spokePool,
+        WETH9Interface _wrappedNativeToken,
+        IPermit2 _permit2
+    ) external nonReentrant {
         if (initialized) revert ContractInitialized();
         initialized = true;
 
         if (!address(_spokePool).isContract()) revert InvalidSpokePool();
```

Comment:
Maybe we should add a check that permit2 is initialized to a real contract similar to the spoke pool address check?